### PR TITLE
include version in manifest's xrefmap

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs [hunk 1]
```diff
@@ -58,7 +59,9 @@ public static Manifest MergeManifest(List<Manifest> manifests)
                 Homepages = (from manifest in manifests
                              from homepage in manifest.Homepages ?? Enumerable.Empty<HomepageInfo>()
                              select homepage).Distinct().ToList(),
-                XRefMap = xrefMaps.Count <= 1 ? xrefMaps.FirstOrDefault() : xrefMaps,
+                XRefMap = xrefMaps.Count == 0
+                    ? null
+                    : xrefMaps.ToDictionary(p => p.Key, p => p.Value),
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: throws exception when key has duplicates?

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -135,16 +135,20 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
                         {
                             Homepages = GetHomepages(context),
                             XRefMap = ExportXRefMap(parameters, context),
+                            XRefMapInfo = new Dictionary<string, XrefMapInfo>
+                            {
+                                [ExportXRefMap(parameters, context)] = new XrefMapInfo
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: ExportXRefMap is called twice

File: src/docfx/SubCommands/DocumentBuilderWrapper.cs [hunk 3]
```diff
@@ -328,7 +328,7 @@ private static IEnumerable<DocumentBuildParameters> ConfigToParameter(BuildJsonC
         private static Dictionary<string, FileMappingParameters> GroupFileMappings(FileMapping content,
             FileMapping overwrite, FileMapping resource)
         {
-            var result = new Dictionary<string, FileMappingParameters>
+            var result = new Dictionary<string, FileMappingParameters>(StringComparer.OrdinalIgnoreCase)
```
Comment (vwxyzh) [inline src/docfx/SubCommands/DocumentBuilderWrapper.cs]: >StringComparer.OrdinalIgnoreCase [](start = 71, length = 32)

remove

File: src/Microsoft.DocAsCode.Plugins/Manifest.cs [hunk 4]
```diff
@@ -36,9 +37,13 @@ public Manifest(IEnumerable<ManifestItem> files)
         [JsonProperty("source_base_path")]
         public string SourceBasePath { get; set; }
 
+        [Obsolete("Please use xrefmap_info")]
         [JsonProperty("xrefmap")]
         public object XRefMap { get; set; }
 
+        [JsonProperty("xrefmap_info")]
+        public Dictionary<string, XrefMapInfo> XRefMapInfo { get; set; }
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Plugins/Manifest.cs]: move to group info.

File: src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs [hunk 5]
```diff
@@ -47,6 +47,10 @@ public static Manifest MergeManifest(List<Manifest> manifests)
             var xrefMaps = (from manifest in manifests
                             where manifest.XRefMap != null
                             select manifest.XRefMap).ToList();
+            var xrefMapsInfo = (from manifest in manifests
+                            where manifest.XRefMapInfo != null
+                            from pair in manifest.XRefMapInfo
+                            select pair).ToList();
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/ManifestUtility.cs]: >pair [](start = 35, length = 4)

distinct pair.key?

**Comment by superyyrrzz** [comment]
plan to add a new key "groups" in manifest, including xrefmap info

**Pull request closed without merging**